static/third/bootstrap-tooltip/tooltip.js

```diff
@@ -19,6 +19,27 @@
  * ========================================================== */
 
 
+/*
+  Function to find whether the device is touch-enabled or not. 
+  Important to disable the tooltip for touch-based devices and allow
+  uninterrupted user events.
+*/ 
+function isTouchScreen(){
+  var result = false
+  if (window.PointerEvent && ('maxTouchPoints' in navigator)) {
+    if (navigator.maxTouchPoints > 0) {
+      result = true
+    }
+  } else {
+    if (window.matchMedia && window.matchMedia("(any-pointer:coarse)").matches) {
+      result = true
+    } else if (window.TouchEvent || ('ontouchstart' in window)) {
+      result = true
+    }
+  }
+  return result
+}
```

Comment:
Is this code borrowed from somewhere?

Reply:
Yes , actually I found the function after looking for ways to detect touch-enabled devices. I made few changes to it and applied to our case.

Reply:
Where did you find it?  We generally cite sources at least in a comment for any code we borrowed from elsewhere.

Reply:
I found it at https://patrickhlauke.github.io/touch/touchscreen-detection/. I will surely mention it in the above comments.

Reply:
Hmm-- that page doesn't have licensing information -- if we're going to use it verbatim, we should try to find licensing details or contact the author for permission.

Reply:
(Or find an alternative implementation that does the same thing -- surely there must be an NPM package that does this).

Reply:
Yes there is an npm package [detect-touch](https://www.npmjs.com/package/detect-touch) which does the same but after the valuable comments posted by @gnprice and @amanagr , I have opened up #16918 which makes use of media query to detect the possibility of hover rather than detection of touch ability itself.

Reply:
It is an approach based on the fact that we want to hide tooltip when hover persist until next click and is activated on click for touch events.